src/ows/src/main/java/org/geoserver/ows/Dispatcher.java

```diff
@@ -500,6 +503,29 @@ BufferedReader fileItemReader(FileItem fileItem) throws IOException {
         return RequestUtils.getBufferedXMLReader(fileItem.getInputStream(), XML_LOOKAHEAD);
     }
 
+    /** parses the AcceptVersions from the request KVP and puts it in the request */
+    void handleAcceptVersionsKVP(Request req) {
+        if ((req.getVersion() != null)
+                || (req.getService() == null)
+                || (req.getRequest() == null)) {
+            return;
+        }
+        if (req.getService().isEmpty() || !req.getRequest().equalsIgnoreCase("getCapabilities")) {
+            return;
+        }
+        String acceptVersions = (String) req.getRawKvp().get("AcceptVersions");
+        if (acceptVersions == null) {
+            return;
+        }
+
+        List<Version> acceptVersionsList =
+                Arrays.asList(acceptVersions.split(",")).stream()
+                        .map(x -> new Version(normalizeVersion(x)))
+                        .collect(Collectors.toList());
```

Comment:
Do you need to sort this list, or is it documented to be in the order preferred by the client?

Reply:
yes - the order specified in the request is important.